packages/liveblocks-yjs/src/index.ts

```diff
@@ -173,6 +214,18 @@ export class LiveblocksYjsProvider<
     return this.rootDocHandler.synced;
   }
 
+  public getStatus(): SyncStatus {
+    if (!this.roomConnected) {
+      return SyncStatus.NotLoaded;
+    }
+    if (!this.synced) {
+      return SyncStatus.Loading;
+    }
+    return this.pending.length === 0
+      ? SyncStatus.Synchronized
+      : SyncStatus.Synchronizing;
+  }
+
```

Comment:
Let's make sure we follow the same convention as we do with `getStorageStatus()` here.

This is the goal behavior as a state diagram:

```mermaid
flowchart LR
    Not-Loaded --> Loading --> Synchronizing --> Synchronized --> Synchronizing
```

Not 100% clear on how this works internally, but I assume this can be in `Not-Loaded` state only during the constructor, but since at the end of the constructor the synchronization seems to start due to [the `this.syncDoc()` call](https://github.com/liveblocks/liveblocks/blob/c93c119cc19627d900b1bd964bcb39076e67e360/packages/liveblocks-yjs/src/index.ts#L133) that should already immediately put this status into the `Loading` state, so `Not-Loaded` should not even be observable for this Yjs provider I think. Therefore I think you can safely remove the `roomConnected` boolean.

So how to decide between `Loading` (initial sync needs to happen) and `Synchronizing`? Could we use something like this?

```suggestion
  public getStatus(): SyncStatus {
    if (this.synced) {
      return SyncStatus.Synchronized;
    }
    
    if (hasEverBeenSynched) {
      return SyncStatus.Synchronizing;
    } else {
      return SyncStatus.Loading;
    }
  }

```

I'm even wondering if you need to do the explicit `pending` tracking here. But maybe I'm missing something here! It looks to me like you could simply set `hasEverBeenSynched = true` when the initial (any, really) sync event happens.

Reply:
We do need to track updates because we currently have no acknowledgement from the server that an update was applied to the source of truth. Most real world issues are pretty well masked by the fact that a full sync happens on reconnect, but if you send off some updates while your internet is farting and close your browser, then those updates will be lost.

Reply:
Oh I see. Fair point. Well, in that case, I think you just need to change the condition of this example to use the `pending` array, I think?

Reply:
What if the room gets disconnected or subdoc is loading? Should that be considered syncronizing?

Reply:
Subdocs I'm not sure about, but when the room gets disconnected and yet the client would have pending changes, the status should still be "synchronizing". To me "synchronizing" means that the client and server are no longer in sync at the moment. The client has changes that the server must be made aware of.

Reply:
This is also how Storage does it.

Reply:
Regarding subdocs, my instinct is to try to treat this the same way. So if the main doc, or _any_ of its subdocs have known pending changes, then the status would be synchronizing. Only if the main doc, and all subdocs, got a synched event, then the entire status should be synchronized. So that the meaning of synchronized means that the client has no changes that need to be submitted.

Another way to think of it is that it's safe to close your browser tab without losing data.

Reply:
I removed roomConnected and not-loaded, but I want to keep loading intact for when the 2-stage sync happens because some editors (like lexical) rely on waiting for sync to complete to show the editor. If we used "synchronizing" for that, then the editor would constantly be flickering in/out, so I think having an explicit loading during initial sync makes more sense. It's a common use case for Yjs editors to wait until sync to allow editing because even though it's technically "okay" to edit during that stage, it looks/feels weird as content pops in.

Reply:
> I think having an explicit loading during initial sync makes more sense

Absolutely – agreed!